public/components/custom_panels/helpers/utils.tsx

```diff
@@ -397,9 +479,7 @@ export const parseSavedVisualizations = (
     timeField: visualization.savedVisualization.selected_timestamp.name,
     selected_date_range: visualization.savedVisualization.selected_date_range,
     selected_fields: visualization.savedVisualization.selected_fields,
-    user_configs: visualization.savedVisualization.user_configs
-      ? JSON.parse(visualization.savedVisualization.user_configs)
-      : {},
```

Comment:
This was added for panels parser. Can you check is panels cypress works as expected?

Reply:
Looking.  But we should not be coding to accomodate testing.   Test suite should (and will) be coded to accomodate production code.

Reply:
Yes, tests are clear

Reply:
This was not added to accomodate testing, this was added to support visualization from events. Cypress tests will catch this.

Reply:
Do you know which events?  Which cypress test demonstrates this?

I only found a test that lazily setup test-data this way.    I could not find a test that ensured user-config was _saved_ as JSON